### ISSUE TITLE
Joining group by proposal should not need `PendingApproval`

### ISSUE THREAD

**Issue by Silver-IT** [okTurtles/group-income#1762]
### Problem

Now with the current code, users are redirected to the `/pending-approval` even though they try to join a group by proposal.
This user flow is needed when using the initial invitation link, but the unique invitation is something that is made specially by proposals of group members, and that is something already approved by the proposal creator.
So `/pending-approval` page shouldn't be redirected, and Welcome page should be displayed.

### Solution

Branch off of the `e2e-protocol` branch, and implement the above userflow.

**Comment by Silver-IT** [comment]
@taoeffect, If you have a better userflow in your mind, please share. Thanks

**Comment by corrideat** [comment]
@Silver-IT I agree that the 'pending approval' page is misleading. However, when joining via an invite link (as opposed to a more direct mechanism not yet implemented that happens entirely on-band), this step is necessary.

We could call the page something more generic and adjust the wording, but the reason it (or something like it) needs to be there is that the link only gives you permission to request the group secret keys. You can become a member only after this step is completed.

What I'm thinking is that, with some small-ish changes, we could send the `inviteAccept` together with the `OP_KEY_REQUEST`.
